graduation-project-tips.md

```diff
@@ -0,0 +1,93 @@
+
+
+## Tip #4: Plan Wisely :hourglass_flowing_sand:
+
+The real work starts from here ! A mistake a lot make is to follow random patterns when working with no clear idea of what they're going to finish in how much time or deadlines whatsoever. **Be smart, don't do that !**
+
+To avoid that mess, you should use a proper software methdology for project planning (recommending **Agile/Scrum** because that's what you're propbably gonna use in work) and software life-cycle planning. And if you're not aware what are those, don't worry there are lot of guides & material that are two clicks away from a google search on those topics.
+
+My second tip here is to use a software that manages your project tasks/backlog, most famous options for this are **[JIRA](https://www.atlassian.com/software/jira)**, and **[Trello Boards](https://trello.com/)**. Those software tools will help bump your productivity and keep you organized throughout the year of hard work.
+Again, knowledge of such tools will be a great qualification for work later on, since not all graudates learn about these stuff before they actually start working.
+
+## Tip #5: Learning Before Everything :books:
+
+This one is important because it's easy to forget actually I did myself and needed someone to remind me with it each time. The graduation project is your learning environment, it's analogous to a course that you put it's content/assignments and study it well. 
+What does this mean? 
+
+- **Don't** pick technology stacks because you're already familiar with them. If there's a stack (new to you) but is a better choice, take the chance and learn it. (I only knew 20% of the knowledge I gained after the year)
+- **Don't** rush tasks, this isn't a product that you're so eager to release then take care later of fine details, rather take your time and perfect each task you work on.
+- **Do** *code peer reviewing*, one of the best ways to learn and get better is to exchange the experience and knowledge and it also ensures code quality.
+- **Do** *knowledge exchange sessions* with each other every now and then to sync everyone's knowledge especially if you're organized as sub-teams.
+
+## Tip #6: Document Everything (literally) :pencil2:
+
+You'll need to document everything that happens in the project life-time. It's a one-year project, don't expect to never forget stuff, on the contrary, you'll always forget details especially finer ones.
+What to document ?
+
+- Requirements Analysis, and project-planning details.
+- Supervisor notes in each meeting with him/her (Meeting Notes).
+- How did you reach an agreement on something (Brainstorming sessions).
+- Links/resources you find along the way (Bugs do re-appear).
+
+## Tip #7: Manage Code Smartly :octocat:
```

Comment:
Might be better to link to an external guide here for clean git usage and git tips. Just to make this guide a bit shorter.

Reply:
Added am external guide. Here it is https://nvie.com/posts/a-successful-git-branching-model/